lib/companies_house/not_found_error.rb

```diff
@@ -4,8 +4,8 @@ module CompaniesHouse
   # Specific error class for when a company cannot be found (for example, if the company
   # number given is invaid)
   class NotFoundError < APIError
-    def initialize(company_id = nil, response = nil)
-      super("Company #{company_id || 'nil'} not found", response)
+    def initialize(resource_id = nil, response = nil)
+      super("Resource #{resource_id || 'nil'} not found", response)
```

Comment:
Would it be worth passing in the resource type here, now it can be multiple things? (at least theoretically!)